port/unix/omrsysinfo.c

```diff
@@ -7557,3 +7559,81 @@ omrsysinfo_get_number_context_switches(struct OMRPortLibrary *portLibrary, uint6
 	return OMRPORT_ERROR_SYSINFO_NOT_SUPPORTED;
 #endif /* defined(LINUX) */
 }
+
+/**
+ * Get the process ID and commandline for each process.
+ * @param[in] portLibrary The port library.
+ * @param[in] callback The function to be invoked for each process with the process ID and command info.
+ * @param[in] userData Data passed to the callback.
+ * @return 0 on success, or the first non-zero value returned by the callback.
+ */
+uintptr_t
+omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallback callback, void *userData)
+{
+#if defined(LINUX)
+	DIR *dir = NULL;
+	struct dirent *entry = NULL;
+	char cmdline_path[PATH_MAX];
+	char comm_path[PATH_MAX];
+	char command[PATH_MAX];
+	uintptr_t file = 0;
+	uintptr_t callback_result = 0;
+	size_t bytes_read = 0;
+	dir = opendir("/proc");
+	if (NULL == dir) {
+		int32_t rc = findError(errno);
+		portLibrary->error_set_last_error(portLibrary, errno, rc);
+		Trc_PRT_sysinfo_get_open_file_count_failedOpeningProcFS(rc);
+		return (uintptr_t)-1;
+	}
+	entry = readdir(dir);
+	while (NULL != entry) {
+		/* Skip non-numeric entries which are not processes */
+		if (0 == isdigit(entry->d_name[0])) {
+			entry = readdir(dir);
+			continue;
+		}
```

Comment:
Suggest moving the call to `readdir()` into the loop (so there need only be one).
Using `strtoull()` earlier, we don't need `isdigit()` (and can validate the whole name).
```c
	for (;;) {
		struct dirent *entry = readdir(dir);
		char *end = NULL;
		size_t bytes_read = 0;
		if (NULL == entry) {
			break;
		}
		if ('\0' == entry->d_name[0]) {
			/* Skip entries with no name. */
			continue;
		}
		pid = (uintptr_t)strtoull(entry->d_name, &end, 10);
		if ('\0' != *end) {
			/* Skip non-numeric entries which are not processes. */
			continue;
		}
```